File: Java-Kotlin/src/main/java/nowcoder/nc106.java
package nowcoder;

import java.util.Arrays;

// sol: https://leetcode-cn.com/problems/maximum-product-of-three-numbers/solution/bu-fei-nao-zi-du-neng-kan-dong-de-jie-xi-69bd/
public class nc106 {

    public long solve(int[] A) {
        // write code here
        if (A == null || A.length < 3) {
            return -1;
        }
        Arrays.sort(A);
        int length = A.length;
        return Math.max((long) A[length - 1] * A[length - 2] * A[length - 3], (long) A[0] * A[1] * A[length - 1]);
    }
}

File: Go-Solutions/nowcoder/nc106.go
package nowcoder

import "sort"

// sol: https://leetcode-cn.com/problems/maximum-product-of-three-numbers/solution/bu-fei-nao-zi-du-neng-kan-dong-de-jie-xi-69bd/

func solve106(A []int) int64 {
	// write code here
	if len(A) < 3 {
		return -1
	}
	sort.Ints(A)
	length := len(A)
	return max106(int64(A[length-1]*A[length-2]*A[length-3]), int64(A[0]*A[1]*A[length-1]))
}

func max106(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}
